### PR TITLE
chore!: upgrade falco to 2.4.2

### PR DESCRIPTION
**Pull request by sumo-drosiek**
Signed-off-by: Dominik Rosiek <drosiek@sumologic.com>

##### Description

Fill in your description here.

---

##### Checklist

- [ ] Changelog updated

###### Testing performed

- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: deploy/helm/sumologic/values.yaml [hunk 1]
```diff
@@ -4265,42 +4265,23 @@ falco:
...
-          done
-      volumeMounts:
-        - mountPath: /host/usr
-          name: usr-fs
-          readOnly: true
-        - mountPath: /host/etc
-          name: etc-fs
-          readOnly: true
+    pullSecrets: []
+
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:4269]:
```suggestion
    # pullSecrets: []

```

File: deploy/helm/sumologic/values.yaml [hunk 2]
```diff
@@ -4267,40 +4267,21 @@ falco:
...
   ## Enable eBPF support for Falco instead of falco-probe kernel module.
   ## Set to true for GKE, for details see:
   ## https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/docs/troubleshoot-collection.md#falco-and-google-kubernetes-engine-gke
   # ebpf:
   #   enabled: true
   falco:
-    jsonOutput: true
+    load_plugins:
+    - json
+    - k8saudit
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:4278]:
```suggestion
      - json
      - k8saudit
```

File: deploy/helm/sumologic/values.yaml [hunk 3]
```diff
@@ -4267,40 +4267,21 @@ falco:
...
-          echo "waiting for kernel headers to be installed"
-          sleep 3
-          done
-      volumeMounts:
-        - mountPath: /host/usr
-          name: usr-fs
-          readOnly: true
-        - mountPath: /host/etc
-          name: etc-fs
-          readOnly: true
```
Comment (swiatekm-sumo) [inline deploy/helm/sumologic/values.yaml:4291]: Why is this not needed anymore?

Comment (sumo-drosiek) [reply deploy/helm/sumologic/values.yaml:4291]: Seems like falco architecture changed a little bit. I wanted to see what e2e tests will show. This configuration works fine for vagrant environment.

No I see that this is likely for openshift only. I will perform manual test then

**Pull request merged**